src/ILCompiler.Compiler/src/Compiler/SymbolReader/PortablePdbSymbolReader.cs

```diff
@@ -0,0 +1,140 @@
+
+        private PortablePdbSymbolReader(MetadataReader reader, MemoryMappedViewAccessor mappedViewAccessor)
+        {
+            _reader = reader;
+            _mappedViewAccessor = mappedViewAccessor;
+        }
+
+        public override IEnumerable<ILSequencePoint> GetSequencePointsForMethod(int methodToken)
+        {
+            var debugInformationHandle = ((MethodDefinitionHandle)MetadataTokens.EntityHandle(methodToken)).ToDebugInformationHandle();
+
+            var debugInformation = _reader.GetMethodDebugInformation(debugInformationHandle);
+
+            var sequencePoints = debugInformation.GetSequencePoints();
+
+            foreach (var sequencePoint in sequencePoints)
+            {
+                if (sequencePoint.StartLine == 0xFEEFEE)
+                    continue;
+
+                var url = _reader.GetString(_reader.GetDocument(sequencePoint.Document).Name);
+
+                yield return new ILSequencePoint() { Document = url, LineNumber = sequencePoint.StartLine, Offset = sequencePoint.Offset };
+            }
+        }
+
+        //
+        // Gather the local details in a scope and then recurse to child scopes
+        //
+        private void ProbeScopeForLocals(List<ILLocalVariable> variables, LocalScopeHandle localScopeHandle)
+        {
+            var localScope = _reader.GetLocalScope(localScopeHandle);
+
+            foreach (var localVariableHandle in localScope.GetLocalVariables())
+            {
+                var localVariable = _reader.GetLocalVariable(localVariableHandle);
+
+                var name = _reader.GetString(localVariable.Name);
+                bool compilerGenerated = (localVariable.Attributes & LocalVariableAttributes.DebuggerHidden) != 0;
+
+                variables.Add(new ILLocalVariable() { Slot = localVariable.Index, Name = name, CompilerGenerated = compilerGenerated });
+            }
+
+            var children = localScope.GetChildren();
+            while (children.MoveNext())
+            {
+                ProbeScopeForLocals(variables, children.Current);
+            }
+        }
+
+        //
+        // Recursively scan the scopes for a method stored in a PDB and gather the local slots
+        // and names for all of them.  This assumes a CSC-like compiler that doesn't re-use
+        // local slots in the same method across scopes.
+        //
+        public override IEnumerable<ILLocalVariable> GetLocalVariableNamesForMethod(int methodToken)
+        {
+            var debugInformationHandle = MetadataTokens.MethodDefinitionHandle(methodToken).ToDebugInformationHandle();
+
+            var localScopes = _reader.GetLocalScopes(debugInformationHandle);
+
+            var variables = new List<ILLocalVariable>();
+            foreach (var localScopeHandle in localScopes)
+            {
+                ProbeScopeForLocals(variables, localScopeHandle);
+            }
+            return variables;
+        }
+    }
```

Comment:
I have added one for completeness, but it is not called. We do not bother with manually closing everything on the compiler shutdown.